packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -864,13 +872,33 @@ export default class API {
     const resp: { statuses: GitHubCommitStatus[] } = await this.request(
       `${this.originRepoURL}/commits/${sha}/status`,
     );
+    const deploys: GitHubDeployments = await this.request(
```

Comment:
Can we do this only if `statuses.length <= 0`?

Reply:
Hey, I implemented it this way to work with `preview_context`, so the end user just looks up the related message to filter on. I think using deploy statuses as a fallback (or commit statuses) would require another config option, since it's entirely possible the user could have different bots leaving messages in both places for different reasons. I don't like the extra requests any more than you do though. I'm open to changing it, or maybe even to just removing the feature from the non-gql API, since I'm assuming it will eventually be retired. What do you think?

Reply:
We can't retire the REST API since it is also used by the `git-gateway` backend.
I does makes sense to check both.